### PR TITLE
KAFKA-12939: After migrating processors, search the codebase for missed migrations

### PR DESCRIPTION
**Pull request by jeqo**
Replacing for new Processor API where `@SuppressWarnings("deprecation") // Old PAPI. Needs to be migrated.` 

There are a few places where Processor Context is used and Processors could not be fully migrated and will require other changes on the codebase:

`DeserializationExceptionHandler#handle`: 
(not sure if this one requires a KIP to change)
```
    @SuppressWarnings("deprecation") // Old PAPI. Needs to be migrated.
    DeserializationHandlerResponse handle(final ProcessorContext context,
                                          final ConsumerRecord<byte[], byte[]> record,
                                          final Exception exception);
```

`ProcessorParameters.oldProcessorSupplier` and `StatefulProcessorNode`:

```
public class ProcessorParameters<KIn, VIn, KOut, VOut> {

    // During the transition to KIP-478, we capture arguments passed from the old API to simplify
    // the performance of casts that we still need to perform. This will eventually be removed.
    @SuppressWarnings("deprecation") // Old PAPI. Needs to be migrated.
    private final org.apache.kafka.streams.processor.ProcessorSupplier<KIn, VIn> oldProcessorSupplier;
```

```
        // temporary hack until KIP-478 is fully implemented
        @SuppressWarnings("deprecation") // Old PAPI. Needs to be migrated.
        final org.apache.kafka.streams.processor.ProcessorSupplier<K, V> oldProcessorSupplier =
            processorParameters().oldProcessorSupplier();
        if (oldProcessorSupplier != null && oldProcessorSupplier.stores() != null) {
            for (final StoreBuilder<?> storeBuilder : oldProcessorSupplier.stores()) {
                topologyBuilder.addStateStore(storeBuilder, processorName);
            }
        }
```

and within tests:

- *TransformTest: include old `ProcessorContext`. This will require a new Transformer API with new PAPI to change.
- ProcessorTopologyTest: contains test to validate old PAPI processors.
- MockProcessor/MockProcessorTest

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by jeqo** [comment]
cc @vvcephei

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordDeserializer.java [hunk 1]
```diff
@@ -70,7 +70,10 @@
         } catch (final Exception deserializationException) {
             final DeserializationExceptionHandler.DeserializationHandlerResponse response;
             try {
-                response = deserializationExceptionHandler.handle(processorContext, rawRecord, deserializationException);
+                response = deserializationExceptionHandler.handle(
+                    (InternalProcessorContext<?, ?>) processorContext,
```
Comment (vvcephei) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordDeserializer.java:74]: Oy. Do you mind filing a `needs-kip` Jira to migrate DeserializationExceptionHandler to the new API?

**Review by vvcephei — APPROVED** [review]
Thanks, @jeqo !

I'm sorry it took me so long to review.

**Comment by vvcephei** [comment]
I've just kicked off a fresh CIT run (https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/4/) to be sure this is still mergeable.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordDeserializer.java (hunk 1 above)
Comment (jeqo) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordDeserializer.java:74]: https://issues.apache.org/jira/browse/KAFKA-13662

Comment (vvcephei) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordDeserializer.java:74]: Thanks!

**Comment by vvcephei** [comment]
Hey @jeqo , it looks like some of those test failures are related to this PR. I took a look at one of them, and it looks like it's due to us trying to cast the processor context, which in that test is just a mock.

**Comment by jeqo** [comment]
My bad. Should be fixed now 👍🏽

**Review by vvcephei — APPROVED** [review]
Thanks!

**Comment by vvcephei** [comment]
There were quite a few test failures:
```
[Build / JDK 17 and Scala 2.13 / kafka.api.PlaintextAdminIntegrationTest.testDescribeReplicaLogDirs()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/6/testReport/junit/kafka.api/PlaintextAdminIntegrationTest/Build___JDK_17_and_Scala_2_13___testDescribeReplicaLogDirs__/)
    [Build / JDK 17 and Scala 2.13 / kafka.server.DynamicBrokerReconfigurationTest.testThreadPoolResize()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/6/testReport/junit/kafka.server/DynamicBrokerReconfigurationTest/Build___JDK_17_and_Scala_2_13___testThreadPoolResize__/)
    [Build / JDK 17 and Scala 2.13 / kafka.server.DynamicBrokerReconfigurationTest.testThreadPoolResize()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/6/testReport/junit/kafka.server/DynamicBrokerReconfigurationTest/Build___JDK_17_and_Scala_2_13___testThreadPoolResize___2/)
    [Build / JDK 17 and Scala 2.13 / org.apache.kafka.streams.examples.wordcount.WordCountDemoTest.testOneWord()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/6/testReport/junit/org.apache.kafka.streams.examples.wordcount/WordCountDemoTest/Build___JDK_17_and_Scala_2_13___testOneWord__/)
    [Build / JDK 17 and Scala 2.13 / org.apache.kafka.streams.examples.wordcount.WordCountDemoTest.testCountListOfWords()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/6/testReport/junit/org.apache.kafka.streams.examples.wordcount/WordCountDemoTest/Build___JDK_17_and_Scala_2_13___testCountListOfWords__/)
    [Build / JDK 17 and Scala 2.13 / org.apache.kafka.streams.examples.wordcount.WordCountDemoTest.testGetStreamsConfig()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/6/testReport/junit/org.apache.kafka.streams.examples.wordcount/WordCountDemoTest/Build___JDK_17_and_Scala_2_13___testGetStreamsConfig__/)
    [Build / JDK 17 and Scala 2.13 / org.apache.kafka.streams.examples.wordcount.WordCountDemoTest.testOneWord()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/6/testReport/junit/org.apache.kafka.streams.examples.wordcount/WordCountDemoTest/Build___JDK_17_and_Scala_2_13___testOneWord___2/)
    [Build / JDK 17 and Scala 2.13 / org.apache.kafka.streams.examples.wordcount.WordCountDemoTest.testCountListOfWords()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/6/testReport/junit/org.apache.kafka.streams.examples.wordcount/WordCountDemoTest/Build___JDK_17_and_Scala_2_13___testCountListOfWords___2/)
    [Build / JDK 17 and Scala 2.13 / org.apache.kafka.streams.examples.wordcount.WordCountDemoTest.testGetStreamsConfig()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/6/testReport/junit/org.apache.kafka.streams.examples.wordcount/WordCountDemoTest/Build___JDK_17_and_Scala_2_13___testGetStreamsConfig___2/)
    [Build / JDK 11 and Scala 2.13 / kafka.admin.DeleteOffsetsConsumerGroupCommandIntegrationTest.testDeleteOffsetsNonExistingGroup()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/6/testReport/junit/kafka.admin/DeleteOffsetsConsumerGroupCommandIntegrationTest/Build___JDK_11_and_Scala_2_13___testDeleteOffsetsNonExistingGroup__/)
    [Build / JDK 11 and Scala 2.13 / kafka.admin.LeaderElectionCommandTest.[1] Type=Raft, Name=testPathToJsonFile, Security=PLAINTEXT](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/6/testReport/junit/kafka.admin/LeaderElectionCommandTest/Build___JDK_11_and_Scala_2_13____1__Type_Raft__Name_testPathToJsonFile__Security_PLAINTEXT/)
    [Build / JDK 11 and Scala 2.13 / kafka.api.ConsumerBounceTest.testClose()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/6/testReport/junit/kafka.api/ConsumerBounceTest/Build___JDK_11_and_Scala_2_13___testClose__/)
    [Build / JDK 11 and Scala 2.13 / kafka.server.DynamicBrokerReconfigurationTest.testThreadPoolResize()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/6/testReport/junit/kafka.server/DynamicBrokerReconfigurationTest/Build___JDK_11_and_Scala_2_13___testThreadPoolResize__/)
    [Build / JDK 11 and Scala 2.13 / kafka.server.DynamicBrokerReconfigurationTest.testThreadPoolResize()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/6/testReport/junit/kafka.server/DynamicBrokerReconfigurationTest/Build___JDK_11_and_Scala_2_13___testThreadPoolResize___2/)
    [Build / JDK 11 and Scala 2.13 / kafka.network.SocketServerTest.remoteCloseWithoutBufferedReceives()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/6/testReport/junit/kafka.network/SocketServerTest/Build___JDK_11_and_Scala_2_13___remoteCloseWithoutBufferedReceives__/)
```

But none of them seem like they could be related. The streams tests failed with:
```
Caused by: org.rocksdb.RocksDBException: While lock file: /tmp/kafka-streams/streams-wordcount/0_0/rocksdb/KSTREAM-AGGREGATE-STATE-STORE-0000000003/LOCK: Resource temporarily unavailable
```

**Comment by vvcephei** [comment]
I'm kicking off one more run to see if it will be clean(er): https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/7/

**Comment by vvcephei** [comment]
Ok, looks like the streams failures were environmental (as they seemed). The others are unrelated:
```
[Build / JDK 11 and Scala 2.13 / kafka.api.ConsumerBounceTest.testClose()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/7/testReport/junit/kafka.api/ConsumerBounceTest/Build___JDK_11_and_Scala_2_13___testClose__/)
    [Build / JDK 8 and Scala 2.12 / kafka.admin.LeaderElectionCommandTest.[1] Type=Raft, Name=testPathToJsonFile, Security=PLAINTEXT](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/7/testReport/junit/kafka.admin/LeaderElectionCommandTest/Build___JDK_8_and_Scala_2_12____1__Type_Raft__Name_testPathToJsonFile__Security_PLAINTEXT/)
    Build / JDK 8 and Scala 2.12 / kafka.admin.LeaderElectionCommandTest.[1] Type=Raft, Name=testElectionResultOutput, Security=PLAINTEXT
    [Build / JDK 8 and Scala 2.12 / kafka.admin.LeaderElectionCommandTest.[1] Type=Raft, Name=testTopicPartition, Security=PLAINTEXT](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/7/testReport/junit/kafka.admin/LeaderElectionCommandTest/Build___JDK_8_and_Scala_2_12____1__Type_Raft__Name_testTopicPartition__Security_PLAINTEXT/)
    [Build / JDK 8 and Scala 2.12 / kafka.admin.LeaderElectionCommandTest.[1] Type=Raft, Name=testPathToJsonFile, Security=PLAINTEXT](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/7/testReport/junit/kafka.admin/LeaderElectionCommandTest/Build___JDK_8_and_Scala_2_12____1__Type_Raft__Name_testPathToJsonFile__Security_PLAINTEXT_2/)
    [Build / JDK 8 and Scala 2.12 / kafka.api.ConsumerBounceTest.testClose()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/7/testReport/junit/kafka.api/ConsumerBounceTest/Build___JDK_8_and_Scala_2_12___testClose__/)
    [Build / JDK 17 and Scala 2.13 / kafka.server.DynamicBrokerReconfigurationTest.testThreadPoolResize()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/7/testReport/junit/kafka.server/DynamicBrokerReconfigurationTest/Build___JDK_17_and_Scala_2_13___testThreadPoolResize__/)
    [Build / JDK 17 and Scala 2.13 / kafka.server.DynamicBrokerReconfigurationTest.testThreadPoolResize()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/7/testReport/junit/kafka.server/DynamicBrokerReconfigurationTest/Build___JDK_17_and_Scala_2_13___testThreadPoolResize___2/)
    [Build / JDK 11 and Scala 2.13 / kafka.server.DynamicBrokerReconfigurationTest.testThreadPoolResize()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/7/testReport/junit/kafka.server/DynamicBrokerReconfigurationTest/Build___JDK_11_and_Scala_2_13___testThreadPoolResize__/)
    [Build / JDK 11 and Scala 2.13 / kafka.server.DynamicBrokerReconfigurationTest.testThreadPoolResize()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-11534/7/testReport/junit/kafka.server/DynamicBrokerReconfigurationTest/Build___JDK_11_and_Scala_2_13___testThreadPoolResize___2/)
    ```

**Comment by vvcephei** [comment]
Thanks so much, @jeqo !

**Pull request merged**